<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"><!-- InstanceBegin template="/Templates/master.dwt" codeOutsideHTMLIsLocked="false" -->
<head>
    <title>CummeRbund - An R package for persistent storage, analysis, and visualization of RNA-Seq from cufflinks output</title>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
	<meta name="description" content="Open source tools for exploration, analysis and visualization of high-throughput RNA-Seq data">
	<link href="css/style.css" rel="stylesheet" type="text/css" />
  	<script type="text/javascript">
  			function recordOutboundLink(link, category, action) {
    		_gat._getTrackerByName()._trackEvent(category, action);
    		setTimeout('document.location = "' + link.href + '"', 100);
  		}
	</script>
    <!-- Place this render call where appropriate -->
    <script type="text/javascript">
      (function() {
        var po = document.createElement('script'); po.type = 'text/javascript'; po.async = true;
        po.src = 'https://apis.google.com/js/plusone.js';
        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(po, s);
      })();
    </script>
    <!--Twitter jscript -->
	<script type="text/javascript" src="//platform.twitter.com/widgets.js"></script>
    <script type="text/javascript">var switchTo5x=false;</script><script type="text/javascript" src="http://w.sharethis.com/button/buttons.js"></script><script type="text/javascript">stLight.options({publisher:'e5ec373d-ef95-4662-abba-95e509cba3c5'});</script>
  </head>
<body>
    <div id="wrap">
      <div id="top">
        <div class="lefts">
          <table cellpadding="2" width="100%">
            <tbody>
              <tr>
                <td> <a href="./index.html">
                    <h1>CummeRbund</h1>
                  </a>
                  <h2>Exploration, analysis and visualization of Cufflinks high-throughput RNA-Seq data</h2>
                </td>
                <td align="right" valign="middle"> 
                 <a href="http://csail.mit.edu/" onClick="recordOutboundLink(this, 'Outbound Links', 'CSAIL');return false;"><img style="height:70px;vertical-align:middle;padding-top:4px" src="images/csail-logo-2C-notext.png" border=0></img></a>&nbsp;
                 <a href="http://www.scrb.harvard.edu/" onClick="recordOutboundLink(this, 'Outbound Links', 'SCRB');return false;"><img style="height:70px;vertical-align:middle;padding-top:4px" src="images/HarvardShield_CMYK.png" border=0></img></a>&nbsp;
               	 
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div id="subheader"><!-- InstanceBeginEditable name="subheader" -->
        <table width="100%">
          <tbody>
            <tr>
              <td><p><strong>CummeRbund</strong> is an R package that is designed to aid and simplify the task of analyzing Cufflinks RNA-Seq output.</p>
                <p>CummeRbund is a collaborative effort between the <a href="http://compbio.mit.edu/">Computational Biology group</a> led by
                  Manolis Kellis at MIT's <a href="http://www.csail.mit.edu/"> Computer Science and Artificial Intelligence Laboratory</a>, and the <a href="http://www.rinnlab.com/">Rinn Lab</a> at the Harvard University department of <a href="http://www.scrb.harvard.edu/">Stem Cells and Regenerative Medicine</a></p>
                <p>CummeRbund is provided under the OSI-approved <a
                  href="http://www.opensource.org/licenses/artistic-license-2.0.php">Artistic License 2.0</a></p></td>
              <td align="right" valign="middle"><a
                  href="http://opensource.org"><img alt="Open Source
                    Software" src="images/osi-certified.gif" border="0" /></a></td>
            </tr>
          </tbody>
        </table>
      <!-- InstanceEndEditable --></div>
      <div id="main">
        <div id="rightside">
          <h2><u>Site Map</u></h2>
          <div class="box">
            <ul>
              <li><a href="index.html">Home</a></li>
              <!-- #NOT SURE THIS IS NEEDED RIGHT NOW <li><a href="tutorial.html">Getting started</a></li> -->
              <li><a href="manual_2_0.html">Manual (Stable)</a> (<a href="manual_2_0.html">HTML</a> | <a href="http://www.bioconductor.org/packages/3.0/bioc/vignettes/cummeRbund/inst/doc/cummeRbund-manual.pdf" target="_blank"  onClick="recordOutboundLink(this, 'Outbound Links', 'Vignette PDF');return false;">PDF</a>)</li>
              <!-- <li><a href="../manual_2_0.html">Manual (Development) </a>(<a href="../manual_2_0.html">HTML</a> | <a href="http://www.bioconductor.org/packages/devel/bioc/vignettes/cummeRbund/inst/doc/cummeRbund-manual.pdf" target="_blank"  onclick="recordOutboundLink(this, 'Outbound Links', 'Vignette PDF 2');return false;">PDF</a>)</li> -->
              <li><a href="http://bioconductor.org/packages/3.0/bioc/manuals/cummeRbund/man/cummeRbund.pdf" target="_blank"  onClick="recordOutboundLink(this, 'Outbound Links', 'Reference PDF');return false;">Reference</a></li>
              <!--<li><a href="cookbook.html">Cookbook</a></li>-->
              <!-- <li><a href="future.html">Future development</a></li> -->
              <li><a href="faq.html">FAQ</a></li>
              <li><!-- Place this tag where you want the +1 button to render --><g:plusone></g:plusone><br />
              <!-- Twitter button
              <a href="https://twitter.com/share" class="twitter-share-button" data-url="http://compbio.mit.edu/cummeRbund/" data-text="CummeRbund: Analysis of Cufflinks RNA-Seq data using #Rstats & #bioconductor" data-count="horizontal">Tweet</a>
              -->
              <!--Share This button-->
              <span  class='st_twitter' ></span><span  class='st_facebook' ></span><span  class='st_email' ></span><span  class='st_delicious' ></span><span  class='st_blogger' ></span><span  class='st_sharethis' ></span>
              </li>
            </ul>
          </div>
          <h2><u>News and updates</u></h2>
          <div class="box">
            <ul>
              <table width="100%">
                <tbody>
                  <tr>
                    <td>New releases and related tools will be announced
                      through the <a
                        href="https://lists.sourceforge.net/lists/listinfo/bowtie-bio-announce"><b>mailing
                          list</b></a></td>
                  </tr>
                </tbody>
              </table>
            </ul>
          </div>
          <h2><u>Getting Help</u></h2>
          <div class="box">
            <ul>
              <table width="100%">
                <tbody>
                  <tr>
                    <td>Questions about CummeRbund should be sent to <a
                        href="mailto:tophat.cufflinks@gmail.com"><b>lgoff@csail.mit.edu</b></a>.                      </td>
                  </tr>
                </tbody>
              </table>
            </ul>
          </div>
       
          <h2><u>Releases</u></h2>
          <div class="box">
            <ul>
              
              <p>Development Version (tied to BioC devel cycle)</p>
              <table width="100%">
                <tbody>
                  <tr>
                    <td><a href="http://bioconductor.org/packages/3.0/bioc/html/cummeRbund.html" onclick="recordOutboundLink(this, 'Bioconductor', 'BioC 3.0');return false;">version 2.7.2</td>
                    <td align="right">5/10/2014</td>
                  </tr>
                  <tr>
                    <td><a href="http://www.bioconductor.org/packages/3.0/bioc/src/contrib/cummeRbund_2.7.2.tar.gz" onClick="recordOutboundLink(this, 'Downloads from BioC', 'Source');return false;">&nbsp;&nbsp;&nbsp;Package Source</a></td>
                  </tr>
                  <tr>
                    <td><a href="http://www.bioconductor.org/packages/3.0/bioc/bin/macosx/contrib/3.1/cummeRbund_2.7.2.tgz" onClick="recordOutboundLink(this, 'Downloads from BioC', 'Mac OS X 10.6');return false;">&nbsp;&nbsp;&nbsp;Mac OS X 10.6 binary</a></td>
                  </tr>
                  <tr>
                    <td><a href="http://www.bioconductor.org/packages/3.0/bioc/bin/windows/contrib/3.1/cummeRbund_2.7.2.zip" onClick="recordOutboundLink(this, 'Downloads from BioC', 'Windows');return false;">&nbsp;&nbsp;&nbsp;Windows binary</a></td>
                  </tr>
                </tbody>
              </table>
              
              <p>Stable Version (tied to Bioconductor release cycle)</p>
              <table width="100%">
                <tbody>
                  <tr>
                    <td><a href="http://bioconductor.org/packages/2.11/bioc/html/cummeRbund.html" onclick="recordOutboundLink(this, 'Bioconductor', 'BioC 2.11');return false;">version 2.6.1</a> (BioC 2.14)</td>
                    <td align="right">10/2/2012</td>
                  </tr>
                  <tr>
                    <td><a href="http://www.bioconductor.org/packages/release/bioc/src/contrib/cummeRbund_2.6.1.tar.gz" onclick="recordOutboundLink(this, 'Downloads from BioC', 'Source');return false;">&nbsp;&nbsp;&nbsp;Package Source</a></td>
                  </tr>
                  <tr>
                    <td><a href="http://www.bioconductor.org/packages/release/bioc/bin/macosx/contrib/3.1/cummeRbund_2.6.1.tgz" onclick="recordOutboundLink(this, 'Downloads from BioC', 'Mac OS X 10.6');return false;">&nbsp;&nbsp;&nbsp;Mac OS X binary 10.6</a></td>
                  </tr>
                  <tr>
                    <td><a href="http://www.bioconductor.org/packages/release/bioc/bin/macosx/mavericks/contrib/3.1/cummeRbund_2.6.1.tgz" onclick="recordOutboundLink(this, 'Downloads from BioC', 'Mac OS X 10.7');return false;">&nbsp;&nbsp;&nbsp;Mac OS X binary 10.7</a></td>
                  </tr>
                  <tr>
                    <td><a href="http://www.bioconductor.org/packages/release/bioc/bin/windows/contrib/3.1/cummeRbund_2.6.1.zip" onclick="recordOutboundLink(this, 'Downloads from BioC', 'Windows');return false;">&nbsp;&nbsp;&nbsp;Windows binary</a></td>
                  </tr>
                </tbody>
              </table>
           <!-- #Development version link table
              
              <p>Development Version (tied to Bioconductor devel cycle)</p>
              <table width="100%">
                <tbody>
                  <tr>
                    <td>version 2.0.1</td>
                    <td align="right">10/3/2012</td>
                  </tr>
                  <tr>
                    <td><a href="http://www.bioconductor.org/packages/devel/bioc/src/contrib/cummeRbund_1.99.3.tar.gz" onclick="recordOutboundLink(this, 'Downloads from BioC', 'Source');return false;">&nbsp;&nbsp;&nbsp;Package Source</a></td>
                  </tr>
                  <tr>
                    <td><a href="http://www.bioconductor.org/packages/devel/bioc/bin/macosx/leopard/contrib/2.15/cummeRbund_1.99.3.tgz" onclick="recordOutboundLink(this, 'Downloads from BioC', 'Mac OS X');return false;">&nbsp;&nbsp;&nbsp;Mac OS X binary</a></td>
                  </tr>
                  <tr>
                    <td><a href="http://www.bioconductor.org/packages/devel/bioc/bin/windows/contrib/2.15/cummeRbund_1.99.3.zip" onclick="recordOutboundLink(this, 'Downloads from BioC', 'Windows');return false;">&nbsp;&nbsp;&nbsp;Windows binary</a></td>
                  </tr>
                </tbody>
              </table>
              -->
              <!-- #Previous Releases -->
              
              <p>Previous Releases</p>
              <table width="100%">
                <tbody>
                  <tr>
                    <td><a href="http://www.bioconductor.org/packages/2.10/bioc/html/cummeRbund.html" onclick="recordOutboundLink(this, 'Bioconductor', 'BioC 2.10');return false;">version 1.2.0</a> (BioC 2.10)</td>
                    <td align="right">4/12/2012</td>
                  </tr>
                </tbody>
              </table>
      
              <p style="font-size:xx-small;font-style:italic">* <a href="http://www.bioconductor.org/packages/release/bioc/html/cummeRbund.html" onClick="recordOutboundLink(this, 'Outbound Links', 'bioconductor.org/cummeRbund');return false;">CummeRbund</a> is now included as part of the <a href="http://www.bioconductor.org"  onClick="recordOutboundLink(this, 'Outbound Links', 'bioconductor.org');return false;" target="_blank">R/Bioconductor</a> project. The links above point to the R packages directly, and  every effort should be made to use the version provided by Bioconductor as it will ensure compatibility with other Bioconductor packages. Easy installation instructions (with dependencies) can be found <a href="http://www.bioconductor.org/packages/release/bioc/html/cummeRbund.html">here</a>.</p>
            </ul>
          </div>
          <h2><u>Related Tools</u></h2>
          <div class="box">
            <ul>
              <li><a href="http://tophat.cbcb.umd.edu/"  onClick="recordOutboundLink(this, 'Outbound Links', 'TopHat');return false;">TopHat</a>:
                Alignment of short RNA-Seq reads</li>
              <li><a href="http://bowtie.cbcb.umd.edu" onClick="recordOutboundLink(this, 'Outbound Links', 'Bowtie');return false;">Bowtie</a>:
                Ultrafast short read alignment</li>
              <li><a href="http://cufflinks.cbcb.umd.edu" onClick="recordOutboundLink(this, 'Outbound Links', 'Cufflinks');return false;">Cufflinks</a>:
                Transcript assembly, differential expression, and differential regulation for RNA-Seq</li>
              <li><a href="http://monocle-bio.sourceforge.net/" onClick="recordOutboundLink(this, 'Outbound Links','Monocle');return false;">Monocle</a>:
                Differential expression and time-series analysis for single-cell RNA-Seq and qPCR experiments</li>
            </ul>
          </div>
          <!--
          <h2>Publications</h2>
          <div class="box">
            <ul>
              
            </ul>
          </div>
          -->
          <h2><u>Contributors</u></h2>
          <div class="box">
            <ul>
              <li><a href="http://people.csail.mit.edu/lgoff/about.html">Loyal A. Goff</a></li>
              <li><a href="http://www.cs.umd.edu/%7Ecole/">Cole Trapnell</a></li>
              <li><a href="http://www.cbcb.umd.edu/~dakelley/">David Kelley</a></li>
              <!--<li>David Kelley</li>-->
             
            </ul>
          </div>
          <h2><u>Links</u></h2>
          <div class="box">
            <ul>
              <li><a href="http://www.csail.mit.edu/" onClick="recordOutboundLink(this, 'Outbound Links', 'CSAIL');return false;">MIT CSAIL</a></li>
              <li><a href="http://www.scrb.harvard.edu/" onClick="recordOutboundLink(this, 'Outbound Links', 'SCRB');return false;">Harvard SCRB</a></li>
              <li><a href="http://compbio.mit.edu/" onClick="recordOutboundLink(this, 'Outbound Links', 'Kellis Lab');return false;">Kellis Lab</a></li>
              <li><a href="http://www.rinnlab.com/"  onClick="recordOutboundLink(this, 'Outbound Links', 'Rinn Lab');return false;">Rinn Lab</a></li>
            </ul>
          </div>
        </div>
        <!-- End of "rightside" -->
        <div id="leftside">
        <!-- InstanceBeginEditable name="leftside" -->
        <h2>Overview</h2>
        <p>High-throughput sequencing of RNA-fragments is a powerful technique that has many applications, including but not limited to transcript assembly, qantitation, and differential expression analysis. The results of these analyses is often very large data sets with a high degree of relations between various data types and can be somewhat overwhelming. CummeRbund was designed to help simplify the analysis and exploration portion of RNA-Seq data derrived from the output of a differential expression analysis using cuffdiff with the goal of providing  fast and intuitive access to your results.</p>
        <p>CummeRbund takes the various output files from a cuffdiff run and creates a SQLite database of the results describing appropriate relationships betweeen genes, transcripts, transcription start sites, and CDS regions. Once stored and indexed, data for these features, even across multiple samples or conditions, can be retrieved very efficiently and allows the user to explore subfeatures of individual genes, or genesets as the analysis requires. We have implemented numerous plotting functions as well for commonly used visualizations. Check back often as we are constantly updating features.</p>
        <p>&nbsp;</p>
        <h2>Release Schedule</h2><br />
        
        <h3>v2.7.2 - 05/10/14</h3>
        <ul>
          <li>A long overdue update for cummeRbund to add a host of new features and plenty of bug fixes. Please see changelog below for a complete list of feature additions and bug-fixes from v2.0.0</li>
          <li>Developmental release as part of Bioconductor 3.0 dev cycle.</li>
        </ul>
        <br />
        <h4>New Features:</h4>
          <ul>
            <li>Added csNMF() method for CuffData and CuffFeatureSet objects to perform non-negative matrix factorization.  As of now, it's merely a wrapper around the default settings for NMFN::nnmf(), but hope to expand in the future. <em>Note: Does not adjust sparsity of matrices after output, must be done by user as needed.</em></li>
            <li>Added csPie() method for CuffGene objects. Allows for visualization of relative isoform, CDS, and promoter usage proportions as a pie chart by condition (or optionally as stacked bar charts by adding + coord_cartesian() ).</li>
            <li>Added 'method' argument to csCluster and csHeatmap to allow custom distance functions for clustering. Default = "none" = JSdist(). You can now provide a function that returns a 'dist' object on rows of a matrix.</li>
            <li>Added varModel.info tracking for compatibility with cuffdiff >=2.1. Will now find varModel.info file if exists, and incorporate into database.</li>
            <li>dispersionPlot() method added for CuffSet object.  This now appropriately draws from varModel.info and is the preferred visualization for dispersion of RNA-Seq data with cummeRbund.</li>
            <li>Added diffTable() method to CuffData and CuffFeatureSet objects to allow a 'one-table' snapshot of results for all Features (CuffData) or a set of Features (CuffFeatureSet). This table outputs key values including gene name, gene short name, expression estimates and per-comparison fold-change, p-value, q-value, and significance values (yes/no). A convenient 'data-dump' function to merge across several tables.</li>
            <li>Added coercion methods for CuffGene objects to create GRanges and GRangeslist objects (more BioC friendly!). Will work on making this possible on CuffFeatureSet and CuffFeature objects as well.</li>
            <li>Added pass-through to select p.adjust method for getSig (method argument to getSig)</li>
            <li>Added ability to revert to cuffdiff q-values for specific paired-wise interrogations with getSig as opposed to re-calculating new ones (useCuffMTC; default=FALSE) </li>
          </ul>
        <br />
        <h4>Bugfixes:</h4>
          <ul>
            <li>Fixed 'fullnames' argument to cuffData::*Matrix() methods so that it does what it's supposed to do.</li>
            <li>Added 'showPool' argument to fpkmSCVPlot.  When TRUE, empirical mean and standard deviation are determined across all conditions as opposed to cross-replicate. This is set to TRUE anytime you have n&lt;2 replicates per condition.</li>
            <li>Added stat="identity" to expressionBarplot to comply with ggplot 0.9.3 enforcement.</li>
            <li>'labels' argument to csScatter is now working as it's supposed to.  You can pass a vector of 'gene_short_name' identifiers to labels and these will be specifically called out in red text on scatterplot.</li>
            <li>Added repFpkmMatrix() and replicates() methods to CuffFeature objects.</li>
            <li>Removed unnecessary Joins to optimize retrieval speed for several key queries.</li>
            <li>Fixed bug in csVolcano matrix that forced ylimits to be c(0,15)</li>
          </ul>
        <br />
        <h4>Notes:</h4>
          <ul>
            <li>Removed generic for 'featureNames'.  Now appropriately uses featureNames generic from Biobase.  As a consequence, Biobase is now a dependency.</li>
            <li>Added passthrough to as.dist(...) in JSdist(...)</li>
            <li>Added 'logMode' argument to csClusterPlot.</li>
            <li>Added 'showPoints' argument to PCAplot to allow disabling of gene values in PCA plot. If false, only sample projections are plotted.</li>
            <li>Added 'facet' argument to expressionPlot to disable faceting by feature_id.</li>
            <li>shannon.entropy now uses log2 instead of log10 to constrain specificity scores between 0 and 1.  </li>
          </ul>
        <br />
        <br />
        <h3>v2.0.0 - 10/2/12</h3>
        <ul>
          <li><strong><em>Stable release:</em></strong> Please see changelog below for a complete list of feature additions and bug-fixes from v1.2.0.</li>
          <li>Released as part of Bioconductor 2.11 release (10/2/12)</li>
          <li>This is a recommended update for all cummeRbund users.</li>
        </ul>
<p>&nbsp;</p>
        <h3>v1.99.6 - 9/24/12 (BETA)</h3>
        <ul>
          <li>          Final beta preview of cummeRbund 2.0 prior to the official release on October 2nd with Bioconductor 2.11.
          </li>
        </ul><br />
        
        <h4>Notes:</h4>
           <ul>
          	<li>'annotation' and &quot;annotation&lt;-&quot; generics were moved to BiocGenerics 0.3.2. Now using appropriate generic function, but requiring BiocGenerics &gt;= 0.3.2</li>
          	<li>Updated vignette and manual.</li>
           </ul>
        <br />
        <h3>v1.99.5 -9/12/12 (BETA)</h3>
          <h4>Bugfixes:</h4>
          <ul>
          <li>Added replicates argument to csDistHeat to view distances between individual replicate samples.</li>
          <li>Appropriately distinguish now between 'annotation' (external attributes) and features (gene-level sub-features).</li>
          <li>csHeatmap now has 'method' argument to pass function for any dissimilarity metric you desire. You must pass a function that returns a 'dist' object applied to rows of a matrix. Default is still JS-distance.</li>
	    </ul>
           <br />
        <h3>v1.99.3 - 9/19/12 (BETA)</h3>
        
        <h4>New Features:</h4>
         	<ul>
          <li>Added diffTable() method to return a table of differential results broken out by pairwise comparison. (more human-readable)</li>
          <li>Added sigMatrix() method to CuffSet objects to draw heatmap showing number of significant genes by pairwise comparison at a given FDR.</li>
          <li>A call to fpkm() now emits calculated (model-derived) standard deviation field as well.</li>
          <li>Can now pass a GTF file as argument to readCufflinks() to integrate transcript model information into database backend.
          	<ul>
            	<li> Added requirement for rtracklayer and GenomicFeatures packages.</li>
          		<li>You must also indicate which genome build the .gtf was created against by using the 'genome' argument to readCufflinks.</li>
            </ul></li>
          <li>Integration with Gviz:
          	<ul>
          		<li>CuffGene objects now have a makeGeneRegionTrack() argument to create a GeneRegionTrack() from transcript model information</li>
          		<li>Can also make GRanges object</li>
          		<li><em>ONLY WORKS IF YOU READ .gtf FILE IN WITH readCufflinks()</em></li>
            </ul></li>
          <li>Added csScatterMatrix() and csVolcanoMatrix() method to CuffData objects.</li>
          <li>Added fpkmSCVPlot() as a CuffData method to visualize replicate-level coefficient of variation across fpkm range per condition.</li>
          <li>Added PCAplot() and MDSplot() for dimensionality reduction visualizations (Principle components, and multi-dimensional scaling respectively)</li>
          <li>Added csDistHeat() to create a heatmap of JS-distances between conditions. </li>
  			</ul>
       	<br />
          <h4>Bugfixes:</h4>
          <ul>
          <li>Fixed diffData 'features' argument so that it now does what it's supposed to do.</li>
          <li>added DB() with signature(object=&quot;CuffSet&quot;) to NAMESPACE</li>
  		  </ul>
          <br />
          <h4>Notes:</h4>
          <ul>
          <li>Once again, there have been modifications to the underlying database schema so you will have to re-run readCufflinks(rebuild=T) to re-analyze existing datasets.</li>
          <li>Importing 'defaults' from plyr instead of requiring entire package (keeps namespace cleaner).</li>
          <li>Set pseudocount=0.0 as default for csDensity() and csScatter() methods (This prevents a visual bias for genes with FPKM &lt;1 and ggplot2 handles removing true zero values).</li>
          </ul>
        <br />
        <br />
        <h3>1.99.2 release - 5/9/12 (BETA)</h3>
        <p><em><strong>CummeRbund 2.0 has been designed to work in conjunction with Cufflinks 2.0. This release is currently a BETA preview. It is fairly robust, however, use with caution.</strong></em></p>
        <h4> Bugfixes:</h4>
        <ul>
         <li>Fixed bug in replicate table that did not apply make.db.names to match samples table.</li>
         <li>Fixed bug for missing values in *.count_tracking files.</li>
         <li>Now correctly applying make.db.names to *.read_group_tracking files.</li>
         <li>Now correctly allows for empty *.count_tracking and *.read_group_tracking files</li>
        </ul>
        <p>&nbsp;</p>
        <h3>1.99.1 release - 5/4/12 (BETA)</h3>
        <p><em><strong>CummeRbund 2.0 has been designed to work in conjunction with Cufflinks 2.0. This release is currently a BETA preview. It is fairly robust, however, use with caution.</strong></em></p>
        <p>This represents a major set of improvements and feature additions. CummeRbund now incorporates additional information emitted from cuffdiff 2.0 including:<br />
        </p>
        <ul>
      <li> run parameters and information.</li>
            <li> sample-level information such as mass and scaling factors.</li>
            <li> individual replicate fpkms and associated statistics for all features.</li>
            <li> raw and normalized count tables and associated statistics all features.</li>
        </ul>
            <h4>&nbsp;</h4>
            <h4>New Features:</h4>
            <ul>
            <li >Please see updated vignette for overview of new features.</li>
            <li >New dispersionPlot() to visualize model fit (mean count vs dispersion) at all feature levels.</li>
            <li >New runInfo() method returns cuffdiff run parameters.</li>
            <li >New replicates() method returns a data.frame of replicate-level parameters and information.</li>
            <li >getGene() and getGenes() can now take a list of any tracking_id or gene_short_name (not just gene_ids) to retrieve a gene or geneset. </li>
            <li >Added getFeatures() method to retrieve a CuffFeatureSet independent of gene-level attributes.  This is ideal for looking at sets of features
            outside of the context of all other gene-related information (i.e. facilitates feature-level analysis)</li>
            <li >Replicate-level fpkm data now available.</li>
            <li >Condition-level raw and normalized count data now available.</li><li >repFpkm(), repFpkmMatrix, count(), and countMatrix are new accessor methods to CuffData, CuffFeatureSet, and CuffFeature objects.</li>
            <li >All relevant plots now have a logical 'replicates' argument (default = F) that when set to TRUE will expose replicate FPKM values in appropriate ways.</li>
            <li >MAPlot() now has 'useCount' argument to draw MA plots using count data as opposed to fpkm estimates.</li>
        </ul>
  <br />
            <h4>Notes:</h4>
            <ul>
            <li>Changed default csHeatmap colorscheme to the much more pleasing 'lightyellow' to 'darkred' through 'orange'.</li>
            <li>SQLite journaling is no longer disabled by default (The benefits outweigh the moderate reduction in load times).</li>
			</ul>
  <br />
            <h4>Bugfixes:</h4>
            <ul>
            <li>Numerous random bug fixes to improve consistency and improve performance for large datasets.</li>
        </ul>
<p>&nbsp;</p>
        <h3>1.2.0 release - 4/2/12 (Version increment only)</h3>
        <ul>
          <li>This release brings the Stable version in-line with the Development version of cummeRbund. 1.2.0 will represent the current stable release.</li>
          <li>There are no additions or changes between this version and 1.1.5.</li>
        </ul>
        <p>&nbsp;</p>
        <h3>1.1.5 release - 3/20/12</h3>
        <p> This release is recommended for all cummeRbund users. Version 1.1.5 will be available as part of the Bioconductor development version 2.10.<br />
        </p>
         <h4>Bugfixes:</h4>
            <ul>
              <li> Fixed minor bug in database setup that caused instability with cuffdiff --no-diff argument.</li>
              <li> Fixed bug in csDendro method for CuffData objects.</li>
              <li> Finished abrupt migration to reshape2. As a result fixed a bug in which 'cast' was still required for several functions and could not be found. Now appropriately using 'dcast' or 'acast'.</li>
              <li> Fixed minor bug in CuffFeature::fpkmMatrix</li>
              </br>
            </ul>

            <h4>New Features:
            </h4>
            <ul>
              <li> Added MAplot() method to CuffData objects.</li>
            </ul>

        <br />
        <h3>1.1.3 release - 2/19/12</h3>
        This release is recommended for all cummeRbund users. Version 1.1.3 will be available as part of the Bioconductor development version 2.10.<br />
        <h4>New Features:</h4>
            <ul>
              <li> getSig() has been split into two functions: 
                <ul>
                  <li>getSig() now returns a vector of ids (no longer a list of vectors)</li>
                  <li>getSigTable() returns a 'testTable' of binary values indicating whether or not a gene was significant in a particular comparison.</li>
                </ul>
              </li>
              <li> Added ability in getSig() to limit retrieval of significant genes to two provided conditions (arguments x &amp; y). (reduces time for function call if you have a specific comparison in mind a priori)
                <ul>
                  <li>When you specify x &amp; y with getSig(), q-values are recalculated from just those selected tests to reduce impact of multiple testing correction.</li>
                  <li>If you do not specificy x &amp; y getSig() will return a vector of tracking_ids for all comparisons (with appropriate MTC). </li>
                </ul>
              </li>
              <li> You can now specify an 'alpha' for getSig() and getSigTable() [0.05 by default to match cuffdiff default] by which to filter the resulting significance calls.</li>
              <li> Added csSpecificity() method:  This method returns a feature-X-condition matrix (same shape as fpkmMatrix) that provides a 'condition-specificity' score 
                <ul>
                  <li>Defined as 1-(JSdist(p,q)) where p is is the density of expression (probability vector of log(FPKM+1)) of a given gene across all conditions, and q is the unit vector for that condition (ie. perfect expression in that particular condition)</li>
                  <li>Specificity = 1.0 if the feature is expressed exclusively in that condition</li>
                </ul>
              </li>
              <li> Created csDendro() method: This method returns a object of class 'dendrogram' (and plots using grid) of JS distances between conditions for all genes in a CuffData, CuffGeneSet, or CuffFeatureSet object.
                <ul>
                  <li>Useful for identifying relationships between conditions for subsets of features</li>
                </ul>
              </li>
              <li> New visual cues in several plot types that indicates the quantification status ('quant_stat' field) of a particular gene:condition. This information is useful to indicate whether or not to trust the expression values for a given gene under a specific condition, and may provide insight into outlier expression values.
                <ul>
                  <li>This feature can be disabled by setting showStatus=F.</li>
                </ul>
              </li>
              <li> csDensity() is now available for CuffFeatureSet and CuffGeneSet objects.<br />
                <br />
              </li>
            </ul>
            <h4>Bugfixes:</h4>
            <ul>
              <li> Fixed bug in getGenes that may have resulted in long query lag for retrieving promoter diffData. As a result all calls to getGenes should be significanly faster.</li>
              <li> CuffData fpkm argument 'features' now returns appropriate data.frame (includes previously un-reported data fields).</li>
              <li> Replaced all instances of 'ln_fold_change' with the actual 'log2_fold_change'.  Values were previously log2 fold change but database headers were not updated to reflect this.</li>
              <li> Fixed bug that could cause readCufflinks() to die with error when using reshape2::melt instead of reshape::melt.<br />
                <br />
              </li>
            </ul>
            <h4>Notes:</h4>
          <strong><em>***The structure of the underlying database has changed in this version.  As a consequence, you must rebuild existing cuffData.db database files to use new version on existing data. readCufflinks(rebuild=T)***</em></strong>
          <ul>
            <li> A 'fullnames' logical argument was added to fpkmMatrix. If True, rownames for fpkmMatrix will be a concatenation of gene_short_name and tracking_id. This has the added benefit of making row labels in csHeatmap easier to read, as well as preserving uniqueness. </li>
            <li> Slight speed improvements to JSdist (noticeable when using csCluster on large feature sets).</li>
            <li> 'testTable' argument to getSig() has been dropped in lieu of new getSigTable() method.</li>
            <li> Updated vignette</li>
          </ul>

<p>&nbsp;</p>
        <h3>1.1.1 release - 1/23/12</h3>
<p>This release is an important bugfix release and is recommended for all cummeRbund users. Version 1.1.1 will be available as part of the Bioconductor development version 2.10.</p>
    <h4>Bugfixes:</h4>
    <ul>
      <li> fixed issue in which there was no graceful error handling of missing CDS or TSS data in cuffdiff output.</li>
      <li> Fixed issue in which distribution test data (promoters, splicing, relCDS) were not appropriately added to objects on creation.</li>
      <li> Fixed bug that would sometimes cause csBoxplot() to throw an error when log-transforming fpkm data. Also added pseudocount argument.</li>
      <li> Fixed bug that would cause diffData() to return a filtered subset of results by default.</li>
      <li> Adjusted indexing of tables to improve performance on large datasets.</li>
      <li> Fixed bug that caused diffData method to not be registered with CuffFeature and CuffGene objects.<br />
        <br />
      </li>
    </ul>
    <h4>New Features:</h4>
    <ul>
      <li> added getSig method to CuffSet class for rapid retrieval of significant features from all pairwise tests (as a list of IDs). By default the level is 'genes' but any feature level can be queried.</li>
      <li> csCluster now uses Jensen-Shannon distance by default (as opposed to Euclidean)</li>
      <li> Added 'xlimits' argument to csVolcano to constrain plot dimensions.</li>
      <li> Enforced requirement in csVolcano for x and y arguments (as sample names).<br />
        <br />
      </li>
    </ul>
    <h4>Notes:</h4>
    <ul>
      <li> Changed dependency 'reshape' to 'reshape2'</li>
      <li> Changed the default orientation of expressionBarplot() for CuffFeatureSet objects.</li>
      <li> Changed output of csCluster to a list format that includes clustering information. As a result, I created the function csClusterPlot<br />
        to replace the previous default drawing behavior of csCluster.  This allows for stable cluster analysis.</li>
      <li> For consistency, the 'testId' slot for CuffDist objects was renamed to 'idField'.  This brings the CuffDist class in line with the CuffData class.</li>
      <li> CuffGene and CuffGeneSet now include slots for promoter, splicing, and relCDS distribution test results.
        </p>
      </li>
    </ul>
<p>&nbsp;</p>
<h3>1.0.0 release - 10/31/11</h3>
<ul>
  <li>	This is the final release candidate for R/Bioconductor v2.9. No major features changes from 0.99.5.</li>
</ul>
<h3>0.99.5 release - 10/7/11</h3>
        <ul>
          <li>This release fixes required elements for acceptance into R/Bioconductor. As of 0.99.5, cummeRbund has been accepted into Bioconductor. Official release will be 1.0.</li>
        </ul>
<h3>0.99.4 release - 9/16/11</h3>
        <ul>
          <li>This release is a private Beta for developers and testers.  First official release will coincide with R/Bioconductor release schedule and will be 1.0.</li>
        </ul>
        <!-- InstanceEndEditable -->
        </div>
        </div>
      <div id="footer">
        <table cellspacing="15" width="100%">
          <tbody>
            <tr>
              <td> This research was supported in part by an NSF Postdoctoral Fellowship. </td>
              <td align="right"> Administrator: <a
                  href="mailto:lgoff@csail.mit.edu">Loyal Goff</a>.
                Design by <a href="http://www.free-css-templates.com"
                  title="Design by David Herreman">David Herreman</a> </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <!-- Google analytics code -->
		<script type="text/javascript">
        
          var _gaq = _gaq || [];
          _gaq.push(['_setAccount', 'UA-318359-6']);
          _gaq.push(['_trackPageview']);
        
          (function() {
            var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
            ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
            var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
          })();
        
        </script>
    <!-- End google analytics code -->
  </body>
<!-- InstanceEnd --></html>
